Modules/Tests/YosemiteTests/Storage/PersistedProductVariationTests.swift

```diff
@@ -0,0 +1,422 @@
+        #expect(persisted.sku == pos.sku)
+        #expect(persisted.globalUniqueID == pos.globalUniqueID)
+        #expect(persisted.price == pos.price)
+        #expect(persisted.downloadable == pos.downloadable)
+        #expect(persisted.fullDescription == pos.fullDescription)
+        #expect(persisted.manageStock == pos.manageStock)
+        #expect(persisted.stockQuantity == pos.stockQuantity)
+        #expect(persisted.stockStatusKey == pos.stockStatusKey)
+    }
+
+    @Test("PersistedProductVariation toPOSProductVariation maps back with attributes and optional image")
+    func test_variation_toPOSProductVariation_maps_back_including_attributes_and_image() throws {
+        // Given
+        let siteID: Int64 = 6
+        let productID: Int64 = 60
+        let variationID: Int64 = 600
+        let persisted = PersistedProductVariation(
+            id: variationID,
+            siteID: siteID,
+            productID: productID,
+            sku: "SKU",
+            globalUniqueID: "GID",
+            price: "11.00",
+            downloadable: true,
+            fullDescription: "Full",
+            manageStock: false,
+            stockQuantity: nil,
+            stockStatusKey: "outofstock"
+        )
+
+        let varAttrs = [
+            PersistedProductVariationAttribute(productVariationID: variationID, name: "Material", option: "Wool"),
+            PersistedProductVariationAttribute(productVariationID: variationID, name: "Fit", option: "Slim")
+        ]
+        let varImage = PersistedProductVariationImage(
+            id: 601,
+            productVariationID: variationID,
+            dateCreated: Date(timeIntervalSince1970: 2000),
+            dateModified: Date(timeIntervalSince1970: 3000),
+            src: "https://example.com/vi.png",
+            name: "vi",
+            alt: "vai")
+
+        // When
+        let pos = persisted.toPOSProductVariation(
+            attributes: varAttrs.map { $0.toProductVariationAttribute() },
+            image: varImage.toProductImage()
+        )
+
+        // Then
+        #expect(pos.siteID == siteID)
+        #expect(pos.productID == productID)
+        #expect(pos.productVariationID == variationID)
+        #expect(pos.sku == persisted.sku)
+        #expect(pos.globalUniqueID == persisted.globalUniqueID)
+        #expect(pos.price == persisted.price)
+        #expect(pos.downloadable == persisted.downloadable)
+        #expect(pos.fullDescription == persisted.fullDescription)
+        #expect(pos.manageStock == persisted.manageStock)
+        #expect(pos.stockQuantity == persisted.stockQuantity)
+        #expect(pos.stockStatusKey == persisted.stockStatusKey)
+        #expect(pos.attributes.count == 2)
+        #expect(pos.image?.imageID == varImage.id)
+    }
+
+    @Test("ProductVariation with associations fetches attributes and image automatically")
+    func test_product_variation_with_associations_fetches_related_records() throws {
```

Comment:
super nit: similar to above, can add Given/When/Then for consistency (same for `test_product_variation_without_image_returns_nil`).